deployment/update_production.sh

```diff
@@ -29,29 +28,28 @@ echo "Starting update..."
 
 set -x # print executed commands. enable this here to not print the if above.
 
-sudo -H -u $USERNAME git fetch
+git fetch
 
 # Note that apache should not be running during most of the upgrade,
 # since then e.g. the backup might be incomplete or the code does not
 # match the database layout, or https://github.com/e-valuation/EvaP/issues/1237.
-[[ -z "$GITHUB_WORKFLOW" ]] && sudo ./deployment/enable_maintenance_mode.sh
+[[ -z "$EVAP_SKIP_APACHE_STEPS" ]] && sudo ./deployment/enable_maintenance_mode.sh
 
-sudo -H -u "$USERNAME" "$ENVDIR/bin/python" manage.py dumpdata --natural-foreign --natural-primary --all -e contenttypes -e auth.Permission --indent 2 --output "$FILENAME"
+./manage.py dumpdata --natural-foreign --natural-primary --all -e contenttypes -e auth.Permission --indent 2 --output "$FILENAME"
 
 [[ ! -z "$EVAP_SKIP_CHECKOUT" ]] && echo "Skipping Checkout"
-[[ ! -z "$EVAP_SKIP_CHECKOUT" ]] || sudo -H -u "$USERNAME" git checkout origin/release
+[[ ! -z "$EVAP_SKIP_CHECKOUT" ]] || git checkout origin/release
 
-sudo -H -u "$USERNAME" "$ENVDIR/bin/pip" install -r requirements.txt
```

Comment:
Ah, we should probably "reload" the nix environment in some way here. I don't think we can just slap a `nix develop` in here and expect the following commands to run in the correct shell. I suppose we could split the "make a backup" and "update yourself" parts and then run each with `nix develop --run ./the-script`. 

I would postpone the details for another issue

Reply:
I think another requirement is that these scripts work without a `nix` setup, so we needed to do that optionally..

Reply:
Is it? We haven't really talked about the production setup yet I believe. Either way, my idea would be to leave nix out of these scripts but have the overarching script that uses nix and calls the current scripts then

Reply:
We can make a follow-up issue to re-evaluate intended production setup, how we want updates and rollbacks to work, and for testing the new approaches.

Reply:
Created #2303